File: Baekjoon/문자열/10988.py
# https://www.acmicpc.net/problem/10988

import sys
input = sys.stdin.readline

word = input().rstrip()
length = len(word)

for i in range(length // 2):
    if word[i] != word[length - i - 1]:
        print(0)
        break
else:
    print(1)

File: Baekjoon/문자열/10988.cpp
// https://www.acmicpc.net/problem/10988

#include <iostream>
#include <string>
using namespace std;

int main() {
    string s;
    cin >> s;
    int len = s.length();
    int answer = 1;
    for (int i = 0; i < len / 2; i++) {
        if (s[i] != s[len - i - 1]) {
            answer = 0;
            break;
        }
    }
    cout << answer;
    return 0;
}